### PR TITLE
Forest lp endpoints

### PR DESCRIPTION
**Pull request by cazapatamar**

### CODE REVIEW

**Comment by cazapatamar** [comment]
> 



> Please check if an error 500 is it ok in some period querys:
> 
> Cases with error when period=2016-2019
> 
>     * basinSubzones&areaId=1303
> 
>     * areaType=states&areaId=85
> 
>     * areaType=basinSubzones&areaId=2501
> 
> 
> ```
> {
>   "code": 500,
>   "userMsg": "res.sendRaw() accepts only strings or buffers"
> }
> ``

According to new data, these are the options to period param. I will fix the documentation

imagen

File: services/forest/api_doc/forestLP.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @apiDefine s_forest_lp Search > Forest
+ * Information related to forest loss ans persistence in a given geofence.
```
Comment (erikasv) [inline services/forest/api_doc/forestLP.js:3]: typo in `ans`

File: services/forest/src/service/forestLP.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     * '2016-2021'
+     *
+     * @returns {Binary} Image with the geometry
+     */
+    getForestLPLayer: async (areaType, areaId, category, period) => {
+      try {
+        const areaGeom = await restAPI.requestAreaGeometry(areaType, areaId);
+        return ForestLPPersistence.findForestLPLayer(
+          areaGeom.features[0].geometry,
+          rasterForestLPSEKeys(category, period),
```
Comment (erikasv) [inline services/forest/src/service/forestLP.js:65]: why `rasterForestLPSEKeys`? SE are not related here.

File: services/forest/src/route/forestLP.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+const { Router } = require('restify-router');
+
+module.exports = (errorHandler, ForestLPService) => {
+  const router = new Router();
+
+  /**
+   * @apiGroup s_forest_lp
+   * @api {get} /forest/lp ForestLP
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/forest/src/route/forestLP.js:8]: To be consistent with other microservices, i like better change SCI endpoints adding `forest` at the beginning. Both, `lp` ad `sci` represent information about forest. If you agree i'm goint to do this change in this PR and later we manage the front.

[the remark this answers is not in the document]
Comment (erikasv) [reply services/forest/src/route/forestLP.js:8]: Yeah, agree. I'll add the adjustment for SCI endpoints in [642](https://github.com/PEM-Humboldt/biotablero-frontend/issues/642)

[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/forest/src/route/forestLP.js:8]: Great!

File: services/forest/api_doc/forestLP.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @apiDefine s_forest_lp Search > Forest
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/forest/api_doc/forestLP.js]: Done!

File: services/forest/src/route/forestLP.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/forest/src/route/forestLP.js:8]: Done!

File: services/forest/src/route/forestLP.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+   * Values for the forest loss and persistence inside the given environmental authority
+   *
+   * Values calculated for 2000-2005, 2006-2010, 2011-2015, 2016-2021 periods
+   *
+   * @apiParam (Query params) {String|Number} areaType area type
+   * @apiParam (Query params) {String} areaId area id
+   *
+   * @apiSuccess {Object[]} result list of objects with information about forest LP
+   * @apiSuccess {String} result.id period
+   * @apiSuccess {String} result.data data for forest LP divided by categories
```
[the remark this answers is not in the document]
Comment (cazapatamar) [reply services/forest/src/route/forestLP.js]: Done!

File: services/forest/src/service/forestLP.js (hunk 2 above)
Comment (cazapatamar) [reply services/forest/src/service/forestLP.js]: Done!

**Comment by cazapatamar** [comment]
> Other thing, please add the change of endpoints path in the gateway (in a new `feature` branch with a MR to develop)

Done!

**Review by Trjegul84 — APPROVED** [review]

File: services/forest/src/route/forestLP.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+   * Values calculated for 2000-2005, 2006-2010, 2011-2015, 2016-2021 periods
+   *
+   * @apiParam (Query params) {String|Number} areaType area type
+   * @apiParam (Query params) {String} areaId area id
+   *
+   * @apiSuccess {Object[]} result list of objects with information about forest LP
+   * @apiSuccess {String} result.id period
+   * @apiSuccess {Object[]} result.data data for forest LP divided by categories
+   * @apiSuccess {String} result.data.area area of the forest loss and persistence category
+   * @apiSuccess {Number} result.data.key forest loss and persistence category
```
Comment (erikasv) [inline services/forest/src/route/forestLP.js:23]: I think these 2 types are crossed?

Comment (cazapatamar) [reply services/forest/src/route/forestLP.js]: Fixed!

**Review by erikasv — APPROVED** [review]

**Pull request merged**